### DISCUSSION TITLE
Fetch multiple pages concurrently with fetchNextPage?

### DISCUSSION THREAD

**Discussion by johsunds** [TanStack/query#4136] [Q&A]
Hi! 

First of all, thank you for the amazing library! 

I'm using `useInfiniteQuery` and I'm trying to trigger multiple concurrent page fetches using `fetchNextPage` and specifying the `pageParam` manually, and I'm finding that it only results in one page being added to my `data`. Is there a way to work around this?

e.g. If I do the following, I want to find pages 1, 5 and 10 in my `data`

```
fetchNextPage({ pageParam: 1 });
fetchNextPage({ pageParam: 5 });
fetchNextPage({ pageParam: 10 });
```

The use case is for incrementally fetching data for different date intervals (not necessarily in order) for a scheduler-like application. The reason I need to fetch multiple pages is that I sometimes need to fetch non-adjacent pages which requires multiple fetches due to the design of the backend service I'm working with. 

EDIT: I realized just after posting this that I could probably make it work by changing the queryFn to fetch multiple pages, however, I'll keep the post open in case someone has a better solution :)

**Comment by TkDodo** [comment]
what happens currently when you call `fetchNextPage` 3 times in a row with different pageParams?

**Reply by johsunds** [reply]
I ended up implementing this particular query without react-query since it felt like I was trying to make `useInfiniteQuery` do something it wasn't designed to do. From what I recall though, when calling `fetchNextPage` 3 times concurrently with different page params, the behavior was as documented for the `cancelRefetch` parameter:

"`options.cancelRefetch: boolean` if set to `true`, calling `fetchNextPage` repeatedly will invoke fetchPage every time, whether the previous invocation has resolved or not. Also, the result from previous invocations will be ignored. If set to `false`, calling `fetchNextPage` repeatedly won't have any effect until the first invocation has resolved. Default is `true`."
